### PR TITLE
"rebase/merge --abort" instead of "reset --hard"

### PR DESCRIPTION
**Pull request by mstv**
Fixes #9575

## Proposed changes

- "rebase/merge --abort" instead "reset --hard"

### Before

image

image

### After

image

image

## Test methodology 

- manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 1e409ce21c9c668b009cfb7f7f2fd4374ac28ac1
- Git 2.35.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 5.0.12
- DPI 96dpi (no scaling)

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
This may be intentional, you may not want to abort the complete merge/rebase, just for the current conflict

**Comment by mstv** [comment]
> This may be intentional, you may not want to abort the complete merge/rebase, just for the current conflict

You are right. In this case, `git reset --hard` resets less of the rebase/merge than `git rebase --abort` indeed.
In #9575, @meniku seems to have lost uncommitted changes from before the merge attempt.
The button is labeled "Abort rebase" / "Abort merge". So I'd rather expect the behavior implemented in this PR.
Perhaps, we should add another button `Reset` with clearer messages.
Do you use the current button for resetting all changes of a conflicting commit?

**Comment by gerhardol** [comment]
> Do you use the current button for resetting all changes of a conflicting commit?

I exit from the form and look at the changes in the diff tab if the changes are not clean.

Removing the button is an option here. (The name is obviously incorrect.)

File: GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs [hunk 1]
```diff
@@ -464,13 +464,13 @@ private void InitializeComponent()
...
             // 
-            this.Reset.Location = new System.Drawing.Point(3, 96);
-            this.Reset.Name = "Reset";
-            this.Reset.Size = new System.Drawing.Size(140, 25);
-            this.Reset.TabIndex = 6;
-            this.Reset.Text = "Abort";
-            this.Reset.UseVisualStyleBackColor = true;
-            this.Reset.Click += new System.EventHandler(this.Reset_Click);
+            this.Abort.Location = new System.Drawing.Point(3, 96);
+            this.Abort.Name = "Reset";
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs:468]:
```suggestion
            this.Abort.Name = "Abort";
```

**Comment by mstv** [comment]
> > Do you use the current button for resetting all changes of a conflicting commit?

I.e. do you use "reset --hard" from FormResolveConflicts?

> I exit from the form and look at the changes in the diff tab if the changes are not clean.
> 
> Removing the button is an option here. (The name is obviously incorrect.)

Which button do you suggest to remove?

I use the button to abort the entire rebase / merge occasionally.
F.i. if the branch already contains an incompatible earlier version of a PR. Then I must remove the outdated version first.

File: GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs (hunk 1 above)
Comment (mstv) [reply GitUI/CommandsDialogs/FormResolveConflicts.Designer.cs]: Done, of course

**Comment by gerhardol** [comment]
> > > Do you use the current button for resetting all changes of a conflicting commit?
> 
> I.e. do you use "reset --hard" from FormResolveConflicts?
> 
> > I exit from the form and look at the changes in the diff tab if the changes are not clean.
> > Removing the button is an option here. (The name is obviously incorrect.)
> 
> Which button do you suggest to remove?
> 
> I use the button to abort the entire rebase / merge occasionally. F.i. if the branch already contains an incompatible earlier version of a PR. Then I must remove the outdated version first.

In FormResolve, I see three options:
1. Rename current button to Reset (and change _abortCurrentOperation slightly).
2. Remove current Abort completely
3. Add abort commit for both commit (*Skip currently applying*) and all as well as Commit

I have not used the Abort in FormResolve as I normally need to to make more detailed changes.
The options in FormRebase is occasionally sufficient.

I vote for 1.

**Pull request closed without merging**